symtab_builder.cc

```diff
@@ -42,6 +44,13 @@ uintptr_t SymtabBuilder::AddSym(const Syminfo& sym) {
     return index;
 }
 
+// Returns true when the symbol specified with (name, soname, version) is
+// defined.
+// TODO(akawashiro) Is it true? Is there any defined symbols whose st_value is
```

Comment:
I think the intention of this function is what you wrote. However, the implementation looks wrong to me (sorry!). As you wrote in this TODO, I think there are some defined symbols with `st_value==0`. `IsDefined` corresponds to https://github.com/lattera/glibc/blob/895ef79e04a953cac1493863bcae29ad85657ee1/elf/dl-lookup.c#L78 and symbols can be defined even if `st_value` is zero when `sym->st_shndx != SHN_ABS` or `stt != STT_TLS`.

I think https://github.com/shinh/sold/blob/83fdbfa24c34c7cf3a22d7f96bdcd0688c8b0fda/symtab_builder.cc#L78 should be corrected to `IsDefined(sym.sym)`, but I'm not 100% sure

Reply:
OK. I fix it using `IsDefined`.